charmhub/http.go

```diff
@@ -35,8 +36,10 @@ type Transport interface {
 }
 
 // DefaultHTTPTransport creates a new HTTPTransport.
-func DefaultHTTPTransport() *http.Client {
-	return &http.Client{}
+func DefaultHTTPTransport(logger Logger) Transport {
+	return jujuhttp.NewClient(jujuhttp.Config{
+		Logger: logger,
```

Comment:
Given the noise created by trace level logging,  I recently had to bring a controller almost to its knees, should use a Child() here, so it can be disabled.  

Trace level logging in jujuhttp, setup and uses httptrace.ClientTrace on Get().